src/java.base/share/classes/sun/util/cldr/CLDRLocaleProviderAdapter.java

```diff
@@ -175,29 +173,19 @@ public TimeZoneNameProvider getTimeZoneNameProvider() {
 
     @Override
     public Locale[] getAvailableLocales() {
-        Set<String> all = createLanguageTagSet("AvailableLocales");
-        Locale[] locs = new Locale[all.size()];
-        int index = 0;
-        for (String tag : all) {
-            locs[index++] = Locale.forLanguageTag(tag);
+        if (AVAILABLE_LOCALES == null) {
+            AVAILABLE_LOCALES = createLanguageTagSet("AvailableLocales").stream()
+                .map(Locale::forLanguageTag)
+                .toArray(Locale[]::new);
         }
-        return locs;
+        return AVAILABLE_LOCALES;
```

Comment:
This should probably clone the cached array:
```suggestion
        return AVAILABLE_LOCALES.clone();
```

Matching what `JRELocaleProviderAdapter` does[^1], since there’s no guarantee that the result of `getAvailableLocales()` won’t be mutated.

[^1]: https://github.com/openjdk/jdk/blob/6b1169e266b9d21864f886ef574dd64116fa2cb0/src/java.base/share/classes/sun/util/locale/provider/JRELocaleProviderAdapter.java#L430-L439

Reply:
Thanks for your comments. In fact, `clone()` in `JRELocaleProviderAdapter` is not necessary, as it will be subsumed into `Locale.getAvailableLocales()` and other `getAvailableLocales()` methods in locale sensitives services providers where they are defensively cloned. Removed this `clone()` invocation.